streaming/src/main/scala/org/apache/spark/streaming/scheduler/BatchInfo.scala

```diff
@@ -32,7 +33,7 @@ import org.apache.spark.streaming.Time
 @DeveloperApi
 case class BatchInfo(
     batchTime: Time,
-    receivedBlockInfo: Map[Int, Array[ReceivedBlockInfo]],
+    streamIdToNumRecords: Map[Int, Long],
     submissionTime: Long,
     processingStartTime: Option[Long],
     processingEndTime: Option[Long]
```

Comment:
Can you make a method called `numRecords` which returns the sum? This is the same approach taken by @zsxwing in  #5533, so will be easier to merge conflicts later.